server/controllers/messages.coffee

```diff
@@ -185,12 +186,17 @@ contentToBuffer = (req, attachment, callback) ->
     # file in the DS, from a previous save of the draft
     # cache it and pass around
     if attachment.url
-        stream = req.message.getBinary filename, (err) ->
+        fileStream = req.message.getBinary filename, (err) ->
             log.error "Attachment streaming error", err if err
 
-        stream_to_buffer_array stream, (err, buffers) ->
-            return callback err if err
-            callback null, Buffer.concat buffers
+        chunks = []
+        bufferer = new stream.Writable
+        bufferer._write = (chunk, enc, next) ->
+            chunks.push(chunk)
+            next()
+        bufferer.end = ->
+            callback null, Buffer.concat chunks
+        fileStream.pipe bufferer
```

Comment:
Don't waste too much time on it, just makes sure it works and explain everything through comments.